src/AlignmentFixerHelper.php

```diff
@@ -0,0 +1,341 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ */
+final class AlignmentFixerHelper
+{
+    /**
+     * @internal
+     */
+    const ALIGN = 'align';
+
+    /**
+     * @internal
+     */
+    const ALIGN_SINGLE_SPACE = 'align_single_space';
+
+    /**
+     * @internal
+     */
+    const ALIGN_SINGLE_SPACE_MINIMAL = 'align_single_space_minimal';
+
+    /**
+     * @const Placeholder used as anchor for right alignment.
+     */
+    const ALIGNABLE_PLACEHOLDER = "\x2 ALIGNABLE%d \x3";
+
+    /**
+     * Keep track of the deepest level ever achieved while
+     * parsing the code. Used later to replace alignment
+     * placeholders with spaces.
+     *
+     * @var int
+     */
+    private $deepestLevel = 0;
+
+    /**
+     * Level counter of the current nest level.
+     * So one level alignments are not mixed with
+     * other level ones.
+     *
+     * @var int
+     */
+    private $currentLevel = 0;
+
+    /**
+     * @param Tokens                $tokens
+     * @param array<string, string> $toAlign
+     */
+    public function fixAlignment(Tokens $tokens, array $toAlign)
+    {
+        foreach ($toAlign as $tokenContent => $alignStrategy) {
+            if (self::ALIGN !== $alignStrategy && self::ALIGN_SINGLE_SPACE !== $alignStrategy && self::ALIGN_SINGLE_SPACE_MINIMAL !== $alignStrategy) {
```

Comment:
For a simple check against two values I see no need for wrapping into `in_array` by the cost of both creating an array and a function call as both are not needed and less explicit.